### DISCUSSION TITLE
IOSS: 2D exodus vectors as 3D vectors

### DISCUSSION THREAD

**Discussion by wortiz** [sandialabs/seacas#470] [Q&A] [answered]
I'm not very familiar with IOSS. For ParaView 2D exodus files currently don't have Glyph and WarpByVector available for the vector data (like displacement/velocity). 

This is due to the IOSS reader returning 2D vectors and ParaView wanting 3D. The legacy reader in ParaView used to manually add a zero third dimension but that isn't present in the IOSS reader.

Is there any way currently to make the underlying Database return 3D vectors instead of 2D for a 2D exodus file?

I currently have a proof of concept fix for VTK https://gitlab.kitware.com/vtk/vtk/-/merge_requests/11319 but was wondering if this could be avoided.

I was thinking maybe something like the non-scalar Tensor transform but I tried a simple one without much luck and it looks like non-scalar ones are not implemented yet. Could just be user error.

**Comment by gsjaardema** [comment, marked as answer]
It would be possible to do this, not sure of the actual effort required or how to specify which 2D fields are expanded to 3D.  It might also be easier at the ParaView level, so will discuss with them also...  

@spyridon97

**Reply by wortiz** [reply]
Thanks. The previous Exodus reader just did it by hand as well. I believe it was only vector fields. I didn't try testing with tensor fields we still write out 1..N1..N instead of the XYZ form so I'm not sure how ParaView treats these.

I just wasn't sure if this was already available I'm not sure this is something that needs to be built in to IOSS since VTK appears to do this translation by hand in a few of the IO readers.

**Reply by wortiz** [reply]
This was easier to do in VTK and has been merged.
